pkgs/development/tools/misc/nrf-command-line-tools/default.nix

```
@@ -0,0 +1,76 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, udev
+, libusb1
+, segger-jlink
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "linux-amd64";
+      sha256 = "1b69272c704f4526b8560fe3c64b92769626261c2a285258bde550d3dd218873";
+    };
+    aarch64-linux = {
+      name = "linux-arm64";
+      sha256 = "bcad32f9340c60374589e6613aa773dd9255e0cea7c50b9bd83ee47ca170a536";
+    };
+    armv7l-linux = {
+      name = "linux-armhf";
+      sha256 = "7b217b434e675dba4ae3a6f611fb9e15835bbe106fd1a4a4052ae68954111a2e";
+    };
+  };
+
+  platform = supported.${stdenv.system} or (throw "unsupported platform ${stdenv.system}");
+
+  version = "10.23.0";
+
+  url = "https://nsscprodmedia.blob.core.windows.net/prod/software-and-other-downloads/desktop-software/nrf-command-line-tools/sw/versions-${lib.versions.major version}-x-x/${lib.versions.major version}-${lib.versions.minor version}-${lib.versions.patch version}/nrf-command-line-tools-${lib.versions.major version}.${lib.versions.minor version}.${lib.versions.patch version}_${platform.name}.tar.gz";
```

We should deduplicate the code by using a let in here.

---

I am unsure how this could be deduplicated, since the fragments are separated by different characters.

---

I think you could create a single string with one separator and then call [`replaceStrings`](https://noogle.dev/?selected=%22builtins.replaceStrings%22&term=%22replace%22) on it to switch the delimiters.

---

Maybe something like this then?

```suggestion
  url = let
    versionWithDashes = builtins.replaceStrings ["."] ["-"] version;
  in "https://nsscprodmedia.blob.core.windows.net/prod/software-and-other-downloads/desktop-software/nrf-command-line-tools/sw/versions-${lib.versions.major version}-x-x/${versionWithDashes}/nrf-command-line-tools-${version}_${platform.name}.tar.gz";
```